spyre-rag/src/common/llm_utils.py

```diff
@@ -1,10 +1,22 @@
 import requests
+from requests.adapters import HTTPAdapter
 import time
 from concurrent.futures import ThreadPoolExecutor, as_completed
 
 from common.misc_utils import get_logger
 from common.settings import get_settings
 
+POOL_SIZE = 10 
+
+adapter = HTTPAdapter(
+    pool_connections=POOL_SIZE, 
+    pool_maxsize=POOL_SIZE, 
```

Comment:
these values will be interesting to deal with! we need to tune them according to our endpoint we will be talking to, e.g: tokenise/embedding apis can take more load and chat completion can limit to batch size allowed for any specific model(e.g: 32)